### ISSUE TITLE
How to detect programmatically if defaultType needs to be specified when accessing spot and swap market?

### ISSUE THREAD

**Issue by Kinzowa** [ccxt/ccxt#16414]
### ؜

Hello,

I  can't find documentation about the instance's option `defaultType` and `defaultSubType` and was wondering how to detect if they need to be set when accessing markets or when sending an order to spot or swap markets. Some exchanges have the options (Binance, Bybit) but not all (Coinbase, Kucoin, Bitmex). Why and when to use them ?

```
In [4]: ccxt.binance().options['defaultType']
Out[4]: 'spot'

In [5]: ccxt.coinbase().options['defaultType']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-5-05a53e58eafe> in <module>
----> 1 ccxt.coinbase().options['defaultType']

KeyError: 'defaultType'

In [6]: ccxt.kraken().options['defaultType']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-6-6940b7a1c511> in <module>
----> 1 ccxt.kraken().options['defaultType']

KeyError: 'defaultType'

In [7]: ccxt.bitmex().options['defaultType']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-7-3c725d5072a7> in <module>
----> 1 ccxt.bitmex().options['defaultType']

KeyError: 'defaultType'

In [8]: ccxt.bybit().options['defaultType']
Out[8]: 'swap'

```

**Comment by carlosmiei** [comment]
Hello @Kinzowa,

`defaultType` is an option that helps us to disambiguate between markets from different types (spot and swap for example) but with conflicting ids.

With the latest developments, if you always use the unified symbols, you only have to worry about methods that don't require a symbol (`fetchBalance`, for instance) . Most of the time (bybit is the only exception here), it will default to `spot` when it has to choose between the spot and swap. 

In short, you might need to change `defaultType` here and there to get non-spot information when the symbol is not available, but it should be effortless to set it up.

**Comment by Kinzowa** [comment]
Hi, Carlos,

Thanks for your answer.

My understanding is that `defaultType` specify an account's type which give access to markets of that specific type. But I'm still scratching my head why, on some exchanges like Kucoin, markets of different types can be accessed without the `defaultType` parameter. Could you please elaborate if it's related to the exchange internal account structure or something different? 

I remember that FTX had a unique account for spot, swap and futures markets, so that no internal transfer of fund were necessary. However, I see that the `transfer()` method is supported by Kucoin whereas this exchange has no `defaultType`'s option, which is a contradiction to me.

BR,
Kinzowa

**Comment by carlosmiei** [comment]
Hello again @Kinzowa 

As I said `defaultType` is just a little helper on our end it's not directly related to any exchange/account.

Kucoin is a peculiar case because it has completely independent APIs/ API keys/implementations for spot and derivative markets, so it does not need any "help" (`defaultType`). `kucoin` will always return spot info whereas `kucoinfutures´ will always return derivatives-related information. 

Let's say you do this
```Python
bybit.fetch_ticker('BTC/USDT') # spot market
bybit.fetch_ticker('BTC/USDT:USDT') # derivative market
```
⬆️ Here, `defaultType` isn't necessary either because our unified symbols are enough to distinguish between a spot or a swap market, so ccxt knows that in the first call, it should use the `spot` endpoint and the `swap` one in the second call.

Now you run this:
```Python
bybit.fetch_balance()
```
⬆️  This is a bit different, `bybit` has different wallets and the endpoint to fetch them is also different, so ccxt has to decide if it should use the `spot` or `swap` endpoint. Since we don't have any symbol available to disambiguate ccxt will resort to the `defaultType` to make this decision unless you specifically provide a `type` inside `params`.

Example:
```Python
bybit.fetch_balance() # will use the defaultType (`swap` in this case) to disambiguate, so will fetch derivative balances
bybit.fetch_balance({"type":"spot"}) # type is provided, so we don't need to resort to `defaultType` and spot balances will be fetched
```

Let me know if you're still not 100% clear on this!

**Comment by Kinzowa** [comment]
Thank you, it's very helpful. Based on your example I assume that Kucoin has a unique wallet for spot, futures and swap.

**Comment by Kinzowa** [comment]
@carlosmiei Kucoin has a Main Account and a Futures Account (.i.e. wallets), both with different interface to GET the account or wallet balance. So how ccxt decides which account the `fetch_balance()` method should return without a `defaultType`'s option ?

**Comment by carlosmiei** [comment]
@Kinzowa yeah, actually kucoin uses the `defaultType`  there
